### DISCUSSION TITLE
Dual Port Ethernet Bridge Native Posix on Intel NUC

### DISCUSSION THREAD

**Discussion by bob2oneil** [zephyrproject-rtos/zephyr#36103] [Q&A] [answered]
I need to create a radio simulator, where there is a standard Ethernet interface as an RJ-45 as the host data source and sink, and I use a secondary Ethernet interface (usually a USB to Ethernet device) to create the simulated over the air RF interface.

I am in the process of doing that with Zephyr using Native Posix execution using virtual Ethernet interfaces.

But what if I want to try this out on actual hardware, that is use an Intel NUC with 2 Ethernet interfaces (say eth0 and eth1) with Zephyr running on them, but with real Ethernet interfaces?

Would I simply need an Ethernet chipset that matches one of the supported drivers (e.g. Intel Intel(R) PRO/1000 Gigabit Ethernet driver E1000) to do this running native POSIX, or is this simply something that requires a dedicated evaluation board with a pair of RJ-45's?

**Comment by jukkar** [comment, marked as answer]
When using Ethernet with native_posix board, the host will need to create a tap device and Zephyr is then using this tap device (named zeth by default) to communicate with host and external systems. If you want to attach the zeth to a real Ethernet port, you can for example bind them both to a bridge so that they are connected together in L2.

**Reply by bob2oneil** [reply]
Thanks Jukka, I have had success with the 1st modality you described, creating the TAP device on the host, setting up host routing, and communicating with Zephyr that way.  

But what if I want to create a Zephyr build that uses a real hardware RJ-45 directly?  Maybe this is simply not possible using the POSIX Native target running on Linux?  

So imagine an Intel NUC with an RJ-45, running Ubuntu, but with Zephyr running as Native POSIX, and targeting an actual physical NIC on the hardware.

Maybe this not possible, but instead would require a real target board with a driver that matches the RJ-45 hardware on that board.

If that is the case, maybe you can elaborate a bit and educate me on the nuances.

**Reply by jukkar** [reply]
> But what if I want to create a Zephyr build that uses a real hardware RJ-45 directly? Maybe this is simply not possible using the POSIX Native target running on Linux?

If you look at `drivers/ethernet/eth_native_posix_adapt.c`, it can send network packets to any network socket. Because Zephyr expects that the network packets have L2 header in them, using the TAP interface is quite convenient.
You could modify this adaptation so that a real network interface is used instead, just make sure that L2 headers are there when Zephyr receives the Ethernet frames.
